inc/ticket.class.php

```diff
@@ -1849,14 +1849,21 @@ function prepareInputForAdd($input) {
       $rules = new RuleTicketCollection($input['entities_id']);
 
       // Set unset variables with are needed
+      $tmprequester = 0;
       $user = new User();
-      if (isset($input["_users_id_requester"])
-          && !is_array($input["_users_id_requester"])
-          && $user->getFromDB($input["_users_id_requester"])) {
-         $input['users_locations'] = $user->fields['locations_id'];
-         $tmprequester = $input["_users_id_requester"];
-      } else {
-         $tmprequester = 0;
+      if (isset($input["_users_id_requester"])) {
+         if (!is_array($input["_users_id_requester"])
+             && $user->getFromDB($input["_users_id_requester"])) {
+            $input['users_locations'] = $user->fields['locations_id'];
+            $tmprequester = $input["_users_id_requester"];
+         } else {
+            if (is_array($input["_users_id_requester"]) && ($user_id = reset($input["_users_id_requester"])) !== false) {
+               if ($user->getFromDB($user_id)) {
+                  $input['users_locations'] = $user->fields['locations_id'];
+                  $tmprequester = $user_id;
```

Comment:
This line is useless. `$tmprequester` vaiable will never be tested in this case as it will not pass following assertion: https://github.com/glpi-project/glpi/pull/5972/files#diff-007edb12589cf3c8df3121cd7b19028aR1885